internal/storage/storage.go

```diff
@@ -0,0 +1,193 @@
+package storage
+
+import (
+	"database/sql"
+	"fmt"
+	"log"
+	"os"
+)
+
+type Scheduler struct {
+	db *sql.DB
+}
+
+type Task struct {
+	ID      string `json:"id"`
+	Date    string `json:"date,omitempty"`
+	Title   string `json:"title"`
+	Comment string `json:"comment,omitempty"`
+	Repeat  string `json:"repeat,omitempty"`
+}
+
+// Не надумал более логичного решения проблемы, что нам иногда нужны все поля
+// Вне зависимости, пустые они или нет
+// Поэтому костыльный дубль структуры выше без omitempty
+type TaskNoEmpty struct {
+	ID      string `json:"id"`
+	Date    string `json:"date"`
+	Title   string `json:"title"`
+	Comment string `json:"comment"`
+	Repeat  string `json:"repeat"`
+}
+
+// Функция открытия коннекта и создания БД, если её нет
+func NewScheduler(DBPath string) (*Scheduler, *sql.DB) {
+	// Проверим, что файлик с БД существует
+	_, err := os.Stat(DBPath)
+
+	var install bool
+	if err != nil {
+		install = true
+	}
+
+	db, err := sql.Open("sqlite", DBPath)
+	if err != nil {
+		log.Fatal(err) // Если к БД коннекта нет, падаем
+	}
+
+	if install {
+		query_create := `CREATE TABLE scheduler(
+			id INTEGER PRIMARY KEY AUTOINCREMENT,
+			date VARCHAR(255), 
+			title VARCHAR(255),
+			comment VARCHAR(255),
+			repeat VARCHAR(126)
+		);`
+		query_index := `CREATE INDEX IF NOT EXISTS date_scheduler ON scheduler (date)`
+
+		_, err = db.Exec(query_create)
+		if err != nil {
+			log.Fatal(err) // Если БД не смогли создать, падаем
+		}
+
+		_, err = db.Exec(query_index)
+		if err != nil {
+			log.Printf("Проблема с созданием индексов: %s", err) // Если не смогли создать индексы, просто проинформируем
+		}
+	}
+
+	return &Scheduler{db: db}, db
+
+}
+
+// Функция инсерта в БД таски
+func (s *Scheduler) PostTask(task Task) (int, error) {
+	stmt, err := s.db.Prepare("INSERT INTO scheduler(date, title, comment, repeat) values(?,?,?,?)")
+	if err != nil {
+		return 0, err
```

Comment:
Ошибки лучше оборачивать, чтобы быстрее понимать где они произошли. 
(Сейчас для сдачи это не обязательно)
```suggestion
		return 0, fmt.Errorf("error while prepearing post request: %w", err)
```